pyaptly/command.py

```diff
@@ -297,69 +313,50 @@ def order_commands(commands, has_dependency_cb=lambda x: False):
         return scheduled
 
 
-class FunctionCommand(Command):
-    """Repesents a function command.
+class DummyCommand(Command):
+    """Represents a dummy command.
 
-    Is used to resolve dependencies between such commands. This command executes
-    the given function. *args and **kwargs are passed through.
+    Is used to resolve dependencies between commands, but does nothing itself
```

Comment:
Was the `FunctionCommand` only ever used as an intermediate / dependency-control thing? If so, could we maybe even get rid of it entirely?

Reply:
I'm not sure it's trivial/useful to remove this. It's used in `snapshot.py` as `indermediate/intermediate2` Commands and allows to require a certain amount of snapshots to provide the `all-snapshots-rotated` or `all-snapshots-rebuilt` dependencies. Since arbitrary other commands can require this it makes sense to have a command for this as well. No "extra logic" is required, only an extra type... Maybe a rename to `virtualCommand` or the like makes it more clear.
@ganwell do you have any input on this?

Reply:
Just answering the question. I'll read the whole PR later: We needed to reread the state at some point, where you now insert the DummyCommands. That is what FunctionCommand is for. I guess you dealt with that by invalidating the cache somehow.

Reply:
I do not want to think about this in this PR